### PR TITLE
Use a ember protocol for file loading

### PR DESCRIPTION
**Pull request by pichfl**
- No longer requires changing the locationType
- caches file system lookups (simple but should do the job)
- [x] Check if this works with locationType:hash

I split registering the Scheme and the protocol in two intentionally to make it easier for developers to register additional protocols.

### CODE REVIEW

**Comment by anulman** [comment]
Yeah! I like your solution much better than the validations

**Comment by pichfl** [comment]
I recommend to merge #155 and #165 before this, but it should work as expected. My understanding for the testing environment is not good enough to include the same thing in the tests in `node-tests/fixtures/project-with-test-config/tests/electron.js` but if someone points me in the right direction I'll file a separate PR to fix those as well.

**Comment by kevingelion** [comment]
This is great! I was having issues rendering a THREE.js scene and couldn't figure out why last week and felt like it was related to relative asset paths or something and turns out this branch works like a dream. Can package/make successfully on ember 2.8!

**Comment by pichfl** [comment]
I'm still in discussion with @bendemboski about details regarding testing, but I'm happy to hear it works as expected.

File: lib/commands/build.js [hunk 1]
```diff
@@ -147,6 +147,7 @@ module.exports = Command.extend({
     let ee = this.project.pkg.config['ember-electron'] || {};
     let patterns = ee['copy-files'] || [
       'ember-electron/electron.js',
+      'ember-electron/protocol-ember.js',
```
Comment (bendemboski) [inline lib/commands/build.js]: No longer needed, right?

File: package.json [hunk 2]
```diff
@@ -110,6 +111,7 @@
     "ember-electron": {
       "copy-files": [
         "ember-electron/electron.js",
+        "ember-electron/protocol-ember.js",
```
Comment (bendemboski) [inline package.json]: Also no longer needed?

File: lib/commands/build.js (hunk 1 above)
Comment (pichfl) [inline lib/commands/build.js]: Aye.

File: package.json (hunk 2 above)
Comment (pichfl) [inline package.json]: 1 AM artifact.

**Comment by felixrieseberg** [comment]
:rocket:

**Pull request merged**